apps/backend/modules/configs.py

```diff
@@ -20,86 +20,110 @@ def float_range(start: float, stop: float, step=1.0):
         count += 1
 
 
-def generate_list(otherVar, paramspos):
-    if otherVar['type'] == 'integer':
-        step = DEFAULT_STEP_INT if otherVar['step'] == '' or int(otherVar['step']) == 0 else otherVar['step']
-        if otherVar['max'] == otherVar['min']:
-            otherVar['max'] = int(otherVar['max']) + int(step)
-        paramspos.append([(otherVar['name'], i) for i in range(int(otherVar['min']), int(otherVar['max']), int(step))])
-    elif otherVar['type'] == 'float':
-        step = DEFAULT_STEP_FLOAT if otherVar['step'] == '' or float(otherVar['step']) == 0 else otherVar['step']
-        if otherVar['max'] == otherVar['min']:
-            otherVar['max'] = float(otherVar['max']) + float(step)
-        paramspos.append([(otherVar['name'], i) for i in float_range(float(otherVar['min']), float(otherVar['max']), float(step))])
-    elif otherVar['type'] == 'string':
-        paramspos.append([(otherVar['name'], otherVar['default'])])
-    elif otherVar['type'] == 'bool':
-        paramspos.append([(otherVar['name'], val) for val in [True, False]])
-
-
-def generate_config_files(hyperparams, unparsedConstInfo):
-    os.mkdir('configFiles')
-    os.chdir('configFiles')
+def generate_list(param: Parameter, paramName, paramspos):
+    if param.type == ParamType.INTEGER:
+        intParam = IntegerParam(**param.dict())
+        paramspos.append([(paramName, i) for i in range(intParam.min, intParam.max, intParam.step)])
+    elif param.type == ParamType.FLOAT:
+        floatParam = FloatParam(**param.dict())
+        paramspos.append([(paramName, i) for i in float_range(floatParam.min, floatParam.max, floatParam.step)])
+    elif param.type == ParamType.BOOL:
+        paramspos.append([(paramName, val) for val in [True, False]])
+
+
+def generate_config_files(experiment: ExperimentData):
+    hyperparams = experiment.hyperparameters
     configIdNumber = 0
     constants = {}
-    parameters = []
-    configs = []
+    parameters = {}
     gather_parameters(hyperparams, constants, parameters)
 
-    for defaultVar in parameters:
+    configDict = {}
+    for defaultKey, defaultVar in parameters.items():
         print(f'Keeping {defaultVar} constant')
-        try:
-            paramspos = []
-            default = [(defaultVar['name'], defaultVar['default'])]
-            paramspos.append(default)
-        except KeyError as err:
-            print(f"Some sorta error with accessing default {err}")
-            return None
-        for otherVar in hyperparams:
-            if otherVar['name'] != defaultVar['name']:
-                try:
-                    generate_list(otherVar, paramspos)
-                except KeyError as err:
-                    raise GladosInternalError('Error during list generation') from err
+        paramspos = []
+        default = [(defaultKey, get_default(defaultVar))]
```

Comment:
Added comment to clarify
"Required to do the cross product, since each config is made by doing a cross product of lists of name value pairs the default variable needs to be a single item list so that there is only one possible value for the default variable"